### PR TITLE
Handle doc generation for docs.rs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/gtk-rs/gir/issues/818.

I'm pretty sure I'm forgetting something here though... Well it's late, I'll see that tomorrow...

cc @EPashkin @sdroege

### CODE REVIEW

File: src/codegen/sys/build.rs [hunk 1]
```diff
@@ -26,21 +26,32 @@ fn generate_build_script(w: &mut dyn Write, env: &Env) -> Result<()> {
...
 use std::env;
+#[cfg(not(feature = "docs-rs"))]
 use std::io::prelude::*;
+#[cfg(not(feature = "docs-rs"))]
 use std::io;
+#[cfg(not(feature = "docs-rs"))]
 use std::process;
 
+#[cfg(feature = "docs-rs")]
+fn main() {} // prevent linking libraries to prevent documentation failure
```
Comment (EPashkin) [inline src/codegen/sys/build.rs]: Maybe rephrase this to remove double "prevent"

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks, LGFM
IMHO only possible addition is updating "README.md"

**Comment by GuillaumeGomez** [comment]
No, I just remembered! I need to add in the features the highest version and "dox".

**Comment by GuillaumeGomez** [comment]
Ok, I think it's ready now.

File: src/codegen/sys/cargo_toml.rs [hunk 2]
```diff
@@ -92,19 +92,42 @@ fn fill_in(root: &mut Table, env: &Env) {
...
+        let mut versions = env
             .namespaces
             .main()
             .versions
             .iter()
-            .filter(|&&v| v > env.config.min_cfg_version);
-        versions.fold(None::<Version>, |prev, &version| {
+            .filter(|&&v| v > env.config.min_cfg_version)
+            .collect::<Vec<_>>();
+        let max_version = versions.iter().max().map(|x| *x);
```
Comment (EPashkin) [inline src/codegen/sys/cargo_toml.rs]: Why not get last element in sorted array?

**Comment by EPashkin** [comment]
It is great idea to add highest version, thanks

File: src/codegen/sys/cargo_toml.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs]: Good point!

**Comment by sdroege** [comment]
Shouldn't we by default just add all features apart from `purge-lgpl-docs`? That's the right thing for almost every crate.

**Comment by GuillaumeGomez** [comment]
The `lgpl-*` features aren't part of sys crates and should be provided through the new option. Also, I'm not a big fan of adding all features. For instance, versions "contain" lower versions so it'd just be noise.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks
